corehq/apps/export/forms.py

```diff
@@ -1,139 +1,336 @@
-    case_type = forms.ChoiceField()
+class CreateCaseExportTagForm(forms.Form):
+    """The information necessary to create an export tag to begin creating a
+    Case Export. This form interacts with CreateExportController in
+    list_exports.ng.js
+    """
+    application = forms.CharField(required=False)
+    case_type = forms.CharField(
+        required=False,
+        help_text=mark_safe(
+            '<span ng-show="!!hasNoCaseTypes '
+            '&& !!createForm.application">{}</span>'.format(
+                ugettext_noop(
+                    """Note: This application does not appear to be using
+<a href="https://wiki.commcarehq.org/display/commcarepublic/Case+Management">
+case management</a>."""
+                )
+            ),
+        ),
+    )
 
-    def __init__(self, domain, *args, **kwargs):
-        super(CreateCaseExportForm, self).__init__(*args, **kwargs)
-        apps = get_apps_in_domain(domain)
-        self.fields['application'].choices = ([
-            ('', _('Select Application...')),
-        ] if len(apps) > 1 else []) + [
-            (app._id, app.name) for app in apps
-        ]
-        self.fields['case_type'].choices = [
-            (module.case_type, module.case_type)
-            for app in apps if hasattr(app, 'modules')
-            for module in app.modules
-            if module.case_type
-        ]
+    def __init__(self, *args, **kwargs):
+        super(CreateCaseExportTagForm, self).__init__(*args, **kwargs)
 
         self.helper = FormHelper()
-        self.helper.form_id = "create-export-form"
-        self.helper.form_class = "form-horizontal"
+        self.helper.form_tag = False
+        self.helper.label_class = 'col-sm-2'
+        self.helper.field_class = 'col-sm-10'
 
         self.helper.layout = crispy.Layout(
-            crispy.Fieldset(
-                _('Select Case Type'),
+            crispy.Field(
+                'application',
+                placeholder=_("Select Application"),
+                ng_model="createForm.application",
+                ng_change="updateCaseTypes()",
+                ng_required="true",
+            ),
+            crispy.Field(
+                'case_type',
+                placeholder=_("Select Case Type"),
+                ng_model="createForm.case_type",
+                ng_disabled="!createForm.application",
+                ng_required="true",
+            ),
+        )
+
+
+class FilterExportDownloadForm(forms.Form):
+    _export_type = 'all'  # should be form or case
+
+    _USER_MOBILE = 'mobile'
+    _USER_DEMO = 'demo_user'
+    _USER_ADMIN = 'admin'
+    _USER_UNKNOWN = 'unknown'
+    _USER_SUPPLY = 'supply'
+
+    _USER_TYPES_CHOICES = [
+        (_USER_MOBILE, ugettext_noop("All Mobile Workers")),
+        (_USER_DEMO, ugettext_noop("Demo User")),
+        (_USER_ADMIN, ugettext_noop("Admin User")),
+        (_USER_UNKNOWN, ugettext_noop("Unknown Users")),
+        (_USER_SUPPLY, ugettext_noop("CommCare Supply")),
+    ]
```

Comment:
I wanted it to be distinct from that filter language, and to make it more readable on the angular js side actually have the strings of what the types are, rather than `if user_type == 1`, etc. I'm of the opinion that reusing code that has tangential use cases (rather than exactly equal) for two different parts of the codebase leads to over-complexity when the original tool isn't designed for what the new use case is, and things get added on haphazardly to fit the new bill. The correct thing would be to refactor `HQUserType` to something more reasonable and readable than it is now, but who knows how long the old reports framework will stick around now that we have UCR.